src/python/pants/backend/native/rules/targets.py

```diff
@@ -0,0 +1,230 @@
+
+
+class NativeFatalWarnings(BoolField):
+    alias = "fatal_warnings"
+    default = False
+
+
+class NativeStrictDeps(BoolField):
+    """Whether to include only dependencies directly declared in the BUILD file.
+
+    If this is False, all transitive dependencies are used when compiling and linking native code.
+    """
+
+    alias = "strict_deps"
+    default = False
+
+
+class ToolchainVariantField(StringField):
+    """Whether to use gcc (gnu) or clang (llvm) to compile.
+
+    Note that currently, despite the choice of toolchain, all linking is done with binutils ld on
+    Linux, and the XCode CLI Tools on MacOS.
+    """
+
+    alias = "toolchain_variant"
+
+    @classmethod
+    def compute_value(cls, raw_value: Optional[str], *, address: Address) -> Optional[str]:
+        value = super().compute_value(raw_value, address=address)
+        valid_options = list(ToolchainVariant.__members__.keys())
+        if value not in valid_options:
+            raise InvalidFieldException(
+                f"The {repr(cls.alias)} field in target {address} must be one of {valid_options}, "
+                f"but was {repr(raw_value)}."
+            )
+        return value
+
+
+class NativeCompilerOptionSets(StringSequenceField):
+    alias = "compiler_options_sets"
+
+
+NATIVE_LIBRARY_COMMON_FIELDS = (
+    *COMMON_TARGET_FIELDS,
+    Dependencies,
+    CtypesNativeLibrary,
+    NativeStrictDeps,
+    NativeFatalWarnings,
+    ToolchainVariantField,
+    NativeCompilerOptionSets,
+)
+
+
+class CLibrary(Target):
+    """A C library that is compatible with Python's ctypes."""
+
+    alias = "ctypes_compatible_c_library"
+    core_fields = (*NATIVE_LIBRARY_COMMON_FIELDS, CSources)
+
+
+class CppLibrary(Target):
+    """A C++ library that is compatible with Python's ctypes."""
+
+    alias = "ctypes_compatible_cpp_library"
+    core_fields = (*NATIVE_LIBRARY_COMMON_FIELDS, CppSources)
+
+
+class ConanPackages(PrimitiveField):
+    """The `ConanRequirement`s to resolve into a `packaged_native_library()` target."""
+
+    alias = "packages"
+    value: Optional[Tuple[ConanRequirement, ...]]
+    required = True
+
+    @classmethod
+    def compute_value(
+        cls, raw_value: Optional[Iterable[ConanRequirement]], *, address: Address
+    ) -> Optional[Tuple[ConanRequirement, ...]]:
+        value_or_default = super().compute_value(raw_value, address=address)
+        if value_or_default is None:
+            return None
+        try:
+            ensure_list(value_or_default, expected_type=ConanRequirement)
+        except ValueError:
+            raise InvalidFieldTypeException(
+                address,
+                cls.alias,
+                value_or_default,
+                expected_type="an iterable of `conan_requirement` objects (e.g. a list)",
+            )
+        return tuple(sorted(value_or_default))
+
+
+class ExternalNativeLibrary(Target):
+    """A set of Conan package strings to be passed to the Conan package manager."""
```

Comment:
Why not name it `ConanNativeLibrary` then?

Reply:
Good idea! The original V1 Target was `ExternalNativeLibrary` but I think this makes more sense.